packages/iframe-execution-environment-service/src/IframeExecutionEnvironmentService.ts

```diff
@@ -111,8 +111,10 @@ export class IframeExecutionEnvironmentService
 
   public async terminatePlugin(pluginName: string) {
     const jobId = this.pluginToJobMap.get(pluginName);
-    jobId && this.terminate(jobId);
-    this._removePluginHooks(pluginName);
+    if (!jobId) {
+      throw new Error(`Job with id "${jobId}" not found.`);
+    }
```

Comment:
added this error to more line up with the `terminate` api that errors in the same way

Reply:
I think I was more lenient here in anticipation that `terminatePlugin` might get called after the corresponding job had been deleted for some other reason, but it's best if we can avoid that. Are we reasonably confident that it won't happen unintentionally, as things stand?